### PR TITLE
Adding support for new compile_string interface

### PR DESCRIPTION
**Pull request by felixfontein**
To be merged when getnikola/nikola#2624 is done.

### CODE REVIEW

File: v7/wordpress_compiler/wordpress/wordpress.py [hunk 1]
```diff
@@ -190,7 +190,14 @@ def __formatData(self, data, context, source=None):
             _LOGGER.warning("The post '" + source + "' still contains shortcodes: " + str(left_shortcodes))
         return output
 
+    def compile_string(self, data, source_path=None, is_two_file=True, post=None, lang=None):
+        """New interface. Might be removed at some time."""
```
Comment (Kwpolska) [inline v7/wordpress_compiler/wordpress/wordpress.py]: Oh come on, we are not that unstable. (this docstring could be more helpful anyways)

Comment (felixfontein) [inline v7/wordpress_compiler/wordpress/wordpress.py]: That's not what I wanted to write :)

**Comment by felixfontein** [comment]
Can you also take a look at 3d73684417a9c27665e01aef424bf47791a35d90?

**Comment by Kwpolska** [comment]
Makes sense.

**Comment by felixfontein** [comment]
Good. Then let's merge this as soon as getnikola/nikola#2623 is merged.

**Pull request merged**